fairseq/data/data_utils.py

```diff
@@ -100,12 +100,14 @@ def check_size(idx):
 
     ignored = []
     itr = collect_filtered(check_size, indices, ignored)
+
+    if len(ignored) > 0 and raise_exception:
+        raise Exception((
+            'Size of sample #{} is invalid (={}) since max_positions={}, '
+            'skip this example with --skip-invalid-size-inputs-valid-test'
+        ).format(idx, size_fn(ignored[0]), max_positions))
```

Comment:
This doesn't work, since collect_filtered is a generator, thus `ignored` will only be populated if the very first example is an invalid size.